test/rpc_test.py

```diff
@@ -501,6 +514,44 @@ def test_shutdown(self):
         # it's safe to call shutdown() multiple times
         rpc.shutdown()
 
+    @dist_init(clean_shutdown=False)
+    def test_wait_all_workers(self):
```

Comment:
Can we add a unit test where we don't setup RPC as part of dist_init and do this manually in the unit test:

1. Call init_rpc.
2. Do some work.
3. Call rpc.shutdown()

Also, I'd recommend running the tests we add here about 100 times locally to ensure we don't have any flakiness.

Reply:
Changing as you recommended.

Reply:
@pritamdamania87 

If I repeatedly `init_rpc()`, I got this error.

```
RuntimeError: Container is already initialized! Cannot initialize it twice! (init at caffe2/torch/csrc/distributed/autograd/context/container.cpp:38)
```
Dist Autograd container is not destroyed after shutting down RPC.

Can we just run once for now?

Reply:
@xush6528 Oh I didn't mean to run the test in a loop here. I meant just run it 100 times locally in your console to ensure there is no flakiness.

Reply:
@pritamdamania87 

Tested locally. It's passing and pretty stable.